### ISSUE TITLE
+vats is broken for some states

### ISSUE THREAD

**Issue by joemfb** [urbit/urbit#6768]
As reported in the UF group:

```
~ship:dojo> +vats|
bail: 4
! /cw/~ship/~zod/~2023.8.22..08.57.30..0953
[%dojo-lame %arvo-response]
> +vats
```

This ship is up-to-date (%base at `0v1s.gbsom.15m55.mpfe9.umbfm.3an52.iaujf.n1c2s.mtfj0.dg0ga.2q87f`).

**Comment by sethfork** [comment]
Also having this issue. I remember seeing this could be related to empty desks

**Comment by silnem2** [comment]
I have this issue as well, but I don't think it's directly related to `+vats`. It looks like a desk called `%~zod` was added to my ship at some point, somehow bypassing the `(sane @tas)` desk name validity check in clay:
``` 
> .^((set desk) %cd /=//=)
[n=%~zod l={%groups %base %quorum %landscape %webterm %garden %rumors %trill %pals %blog %portal} r={%kids %talk}]
```

**Comment by bradyab** [comment]
I'm having this as well, i can't nuke or uninstall the zod desk either:

```
> dojo: |nuke %~zod
nest-need [term=atas %~]
dojo: nest-have [%~zod %~7
dojo: bad-argument (supplied sample incorrect)
```

**Comment by mikolajpp** [comment]
This is interesting. A invalid term `%~zod` would definitely not bypass `+sane`
```
> ((sane %tas) '~zod')
%.n
```
`%~zod` -- a constant atom `~zod` would, but it is a desk `%$`.

When you type `|nuke %~zod` in dojo you are actually trying to nuke the empty desk `%$`, as %~zod is not a term. It is 0 and gets cast down to a `@tas`
```
> `@tas`%~zod
%$
```

When you do a scry to get available desk, there indeed seems to be a desk named '~zod'. To nuke it, one would instead have to 
```
> |nuke '~zod'
```

**Comment by tinnus-napbus** [comment]
fyi `|nuke` operates on agents not desks. You can do `|nuke %the-desk, =desk &` to nuke every agent on a desk but otherwise it's `|nuke %the-agent`

**Comment by silnem2** [comment]
@mikolajpp Yeah, the `(sane %tas)` check was added to `%fuse`, `%park`, and `%merg` in https://github.com/urbit/urbit/pull/6304, but other requests (e.g. `%zest`) can add references to desks that don't exist to clay's map of desks and don't include a sanity check.

**Comment by neonfuz** [comment]
I'm having this issue, does anyone know a fix?

**Comment by bradyab** [comment]
`|nuke '~zod'` worked, but `+vats` is still broken:

```
! /cw/~toptyr-bilder/~zod/~2023.10.8..17.30.09..f918 
[%dojo-lame %arvo-response]
```

**Comment by dolled-possum** [comment]
Encountering this issue in at least two ships in our hosting environment.  In both cases, +vats is broken with the %dojo-lame response above, and in both cases, '~zod' is listed as a desk in the .^((set desk) %cd /=//=) scry.  Importantly, it also wrecks the .^(json %gx /=hood=/kiln/pikes/json) scry, with
`bail: 4

bail: 2
/cz/~<somę ship>/~zod/~2023.10.13..17.20.27..d22a
dojo: failed to process input
peek bad result
/cz/~<somę-ship>/~zod/~2023.10.13..17.20.34..7832`

**Comment by dolled-possum** [comment]
Regarding the third party desks that these ships with broken vats/pikes have in common, here's the relatively short list:
blog
houston
pals
portal
radio
realm
rumors
trill
And naturally, there are plenty of ships with those desks installed that _aren't_ exhibiting these problems, so it's definitely not a sufficient cause, but possibly a necessary one.  Perhaps other people in this thread can further narrow the list.

**Comment by chrisalexadams** [comment]
Same issue for awhile here.
